### PR TITLE
Assignment10: theme change between dark mode and light mode in header, footer and card components using context API

### PR DESCRIPTION
**Pull request by nitishnivedan**
Assignment10: theme change between dark mode and light mode in header, footer and card components using context API

### CODE REVIEW

**Comment by nitishnivedan** [comment]
**On page load**

image

**After click on the Change color button**

image

**Footer** always the **default color** and doesnt get the updated value because it is not **wrapped** inside **<ThemeConext.Provider>**

File: src/components/Header/index.js [hunk 1]
```diff
@@ -1,28 +1,37 @@
...
-  </>
-);
+const Header = () => {
+  const [color, setColor] = useState("white");
+  return (
+    <>
+      <ThemeContext.Provider value={{ color, setColor }}>
+        <div className="header" style={{ backgroundColor: color }}>
+          <div className="headerComponents">
+            <img src={logo} width="50" height="50" />
```
Comment (ap221882) [inline src/components/Header/index.js:17]: can add alt attribute

File: src/components/Root/index.js [hunk 2]
```diff
@@ -52,14 +54,24 @@ const RootElement = () => {
...
         teamInformation={isDataFetched ? fullInfoOfUsers : teamInformation}
       />
-
+      <button
+        style={{
+          height: "53px",
+          width: "20%",
+          marginTop: "1%",
+        }}
+        onClick={() => setColor(color === "white" ? "black" : "white")}
```
Comment (ap221882) [inline src/components/Root/index.js:63]: can use callback inside setstate to avoid dispatch state collisions

**Review by ap221882 — COMMENTED** [review]
Reviewed. Can avoid camelcasing in case of CSS naming. Nice work.

File: src/components/Header/index.js [hunk 3]
```diff
@@ -1,28 +1,37 @@
-import { Outlet } from "react-router-dom";
+import { createPath, Outlet } from "react-router-dom";
```
Comment (smriti-porwal) [inline src/components/Header/index.js:1]: what is the need to add createPath?

**Review by iswarya15 — APPROVED** [review]
Reviewed, looks good.

**Review by nam334 — COMMENTED** [review]
Reviewed!

File: src/components/Header/index.js (hunk 3 above)
Comment (nitishnivedan) [reply src/components/Header/index.js:1]: mistakenly added. not using it anywhere. will remove

**Pull request merged**